src/NuGet.Core/NuGet.Resolver/PackageResolver.cs

```diff
@@ -41,6 +42,34 @@ public IEnumerable<PackageIdentity> Resolve(PackageResolverContext context, Canc
                     throw new NuGetResolverInputException(String.Format(CultureInfo.CurrentCulture, Strings.MissingDependencyInfo, requiredId));
                 }
             }
+            
+            var invalidsPackages = new List<string>();
+
+            var installedPackages = context.PackagesConfig.Select(p => p.PackageIdentity).ToArray();
+
+            // validate existing package.config for any invalid dependency
+            foreach (var package in installedPackages)
+            {
+                var existingPackage = context.AvailablePackages.FirstOrDefault(p => StringComparer.OrdinalIgnoreCase.Equals(p.Id, package.Id) && p.Version.Equals(package.Version));
+
+                if (existingPackage != null)
```

Comment:
there won't be a case there existing package is not found from available packages until someone manually delete that package from package folder or global cache and also remove it's source from the project.

I'll anyways add warning message for this as well in case someone really messes up his project config.